lib/src/rules/duplicated_keys.dart

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2020, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+
+import '../analyzer.dart';
+
+const _desc = r"Two keys in a constant set/map shouldn't be equal.";
```

Comment:
The description says "constant", but neither the examples nor the tests use `const`. So I suspect that you mean "Two constant-valued elements / keys in a set / map literal shouldn't be equal.".